File: app/src/main/java/com/vinicius/productstoreapp/ProductDao.java
package com.vinicius.productstoreapp;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import java.util.List;

@Dao
public interface ProductDao {
    @Insert
    void insert(Product product);

    @Query("SELECT * FROM products")
    List<Product> getAllProducts();
}

File: app/src/main/java/com/vinicius/productstoreapp/ProductDao.kt
package com.vinicius.productstoreapp

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query

@Dao
interface ProductDao {
    @Insert
    suspend fun insert(product: Product)

    @Query("SELECT * FROM products")
    suspend fun getAllProducts(): List<Product>
}
